packages/core/docs/commercetools/migrate/1.4.0/index.md

```diff
@@ -0,0 +1,124 @@
+
+From now on, the application will request one access token for server operations (introduced in the previous release) and one for all guests. The server stores these tokens and doesn't send them out to the users. It will only create session cookies only for anonymous and logged-in users.
+
+:::tip What is an anonymous session in commercetools?
+You might be wondering what an "anonymous" session is and how it's different from the "guest" session. In commercetools there are three types of sessions:
+
+* **guest** - user that didn't log in nor perform any visitor-specific operation,
+* **anonymous** - user that didn't log in but performed some visitor-specific operation, for example, added item to the cart or wishlist,
+* **customer** - user that logged into the application.
+:::
+
+This change by itself doesn't require any changes in existing projects. However, we used this opportunity to improve the security of the cookies storing the user session. See the next section for more details.
+
+### Improving security of the session cookie
+
+The commercetools integration creates the `vsf-commercetools-token` cookie to store the user session. In the past, this cookie didn't include the `secure` and `httpOnly` attributes. The lack of these attributes meant that the cookie could be sent with unsecured HTTP and be accessed using JavaScript, which exposed the cookie to security risks.
+
+From now on, a session cookie will always have the `httpOnly` attribute. It will also have the `secure` attribute, but only if the request was HTTPS. For this reason, you don't have to use HTTPS in your development environments.
+
+However, because the session cookie can no longer be accessed using JavaScript, we needed a new way of checking if the user is logged in. For this reason, we introduced a new `isLoggedIn` endpoint. **You need to update the `middleware/is-authenticated.js` file**, which protects `/my-account` routes from unauthorized access and all other parts of the application that read the session cookie.
+
+```javascript
+// middleware/is-authenticated.js
+import { Logger } from '@vue-storefront/core';
+
+export default async ({ $vsf, route, redirect }) => {
+  try {
+    const isLoggedIn = await $vsf.$ct.api.isLoggedIn();
+
+    if (!isLoggedIn) {
+      throw new Error(`"${ route.fullPath }" route is only available to logged-in customers`);
+    }
+  } catch (error) {
+    Logger.warn(error.toString());
+    return redirect('/');
+  }
+};
+```
+
+In components, you can use the same approach by using the `useVSFContext` helper to access the `$ct` context.
+
+```javascript
+// Vue components
+import { useVSFContext } from '@vue-storefront/core';
+
+export default {
+  setup() {
+    const { $ct } = useVSFContext();
+
+    async function isLoggedIn() {
+      return await $vsf.$ct.api.isLoggedIn();
```

Comment:
`$vsf` is undefined

Reply:
Fixed in https://github.com/vuestorefront/vue-storefront/pull/6512